### DISCUSSION TITLE
How do you consider the interactions between variables during feature extraction?

### DISCUSSION THREAD

**Discussion by levisocool** [blue-yonder/tsfresh#1055] [Q&A]
Hi! Thanks for your team's hard work on the code.
I need some opinion from you about the Feature Extraction procedure. 
For tabular data, e.g. vehicle traveling data, there exists various variables (origin features) like speed, GPS, battery current, battery voltage and so on. In tsfresh, the function Extract_features produces hundreds of new features as the website link describes (https://tsfresh.readthedocs.io/en/latest/text/list_of_features.html). 
However, the Extract_features func only consider transforming on a single variable, for example, sum/mean/coefficient of speed. It did not consider the interactions between variables (origin features). For example, where are high speed situations occouring? (combing speed and GPS two variables). Or the mean value of battery voltage when speed exceeding 70 km/h.
I know the idea describe above would produce numerous new features as tsfresh did. But i still want to know how do you and your team considering about the interactions between variables during feature extraction?
Correct me if my understanding about Extract_features func is wrong. Thanks!

**Comment by nils-braun** [comment]
Your assumption is correct, tsfresh does not consider correlation between different variables/timeseries. It would be a very nice feature addition, but it would also require a major rework of the framework and a lot of additional computing time during the extraction.
If you already know about potential correlations, I would recommend you combine the two original variables into a new time series and feed this into the feature extraction.
